File: perl/arrays/arrays/assignment_and_overwriting.pl
#
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
#

use strict;
use warnings;

my @array = ();

print("Not initialised, before assignments: \@array = ();\n");
print(@array, "\n\n");

@array = (0, 0, 0);

print("Initialised, before assignments: \@array = (0, 0, 0);\n");
print(@array, "\n\n");

$array[1] = 6;

print("After assignment: \$array[1] = 6;\n");
print(@array, "\n\n");

$array[1] = 10;

print("After overwriting: \$array[1] = 10;\n");
print(@array, "\n\n");

File: ruby/arrays/arrays/assignment_and_overwriting.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

array = []

puts("Not initialised, before assignments: array = []\n\n")
puts(array.inspect, "\n")

array = [0, 0, 0]

puts("Initialised, before assignments: array = [0, 0, 0]\n\n")
puts(array.inspect, "\n")

array[1] = 6

puts("After assignment: array[1] = 6\n\n")
puts(array.inspect, "\n")

array[1] = 10

puts("After overwriting: array[1] = 10\n\n")
puts(array.inspect, "\n")
